build.ts

```diff
@@ -269,6 +269,15 @@ class RunningServer {
       await this.killProcessInSlot(slotToStartIn, {drain: false});
     }
     this.startProcessInSlot(slotToStartIn);
+    if (process.env.RESTART_INTERVAL) {
+      const intervalSeconds = parseInt(process.env.RESTART_INTERVAL);
+      const noise = Math.round(intervalSeconds * 0.1 * (Math.random() - 0.5) * 2);
+      const intervalWithNoise = intervalSeconds + noise;
+      console.log(`Scheduling restart in ${intervalWithNoise} seconds (original: ${intervalSeconds} seconds, noise: ${noise} seconds)`);
+      // Note: This restart causes around 10s of downtime for this instance
+      setTimeout(() => this.startOrRestart(), intervalWithNoise * 1000);
```

Comment:
10s of downtime every 30 mins will be a mild irritation to users. In principle the vite version of this script supports keeping the old server running while the new one starts up, we could probably get that to work, but I think it's better to keep it simple and be sure it's working